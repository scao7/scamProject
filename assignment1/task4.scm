;task4
;print the line with the level value
(define (crazyTriangle left right)
  	(println left right)
	(define a left)
	(define b right)
	(define (pascaltriangle n)
  		(define c )
		(define (for-iter1 i)
			(set! a left)
			(set! b right)
			(set! c left 
			(define (for-iter2 j)
				(print " ")
		  		(if( <= j (- n i 1))
					(for-iter2 (+ j 1))
				  
			  )
			(for-iter2 1)
			(define (for-iter3 k)
			;	(print c " " )
				(print c " ")
				(set! c ( /(* c (- i k))  (+ k 1)))
			;	(set! b ( /(* b (- i k))  (+ k 1)))
			
	  			(if(<= k i)
					(for-iter3 (+ k 1))
			  	)
	 	 	)
			(for-iter3 0)
			(println)
			(if( < i n)
  				(for-iter1 (+ i 1))
			)


 		 )
		(for-iter1 0)
	)
)

(define (main)
	(setPort(open (getElement ScamArgs 1) 'read))
	(define arg1 (readExpr))
	(define arg2 (readExpr))
	(define arg3 (readExpr))
	((crazyTriangle arg1 arg2 ) arg3)
  )
