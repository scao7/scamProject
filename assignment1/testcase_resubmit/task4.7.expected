((crazyTriangle -11 -11) 10)
         -11
        -11 -11
       -11 -22 -11
      -11 -33 -33 -11
     -11 -44 -66 -44 -11
    -11 -55 -110 -110 -55 -11
   -11 -66 -165 -220 -165 -66 -11
  -11 -77 -231 -385 -385 -231 -77 -11
 -11 -88 -308 -616 -770 -616 -308 -88 -11
-11 -99 -396 -924 -1386 -1386 -924 -396 -99 -11
